src/dsql/ExprNodes.h

```diff
@@ -955,6 +955,12 @@ class LiteralNode final : public TypedNode<ValueExprNode, ExprNode::TYPE_LITERAL
 	virtual ValueExprNode* pass2(thread_db* tdbb, CompilerScratch* csb);
 	virtual dsc* execute(thread_db* tdbb, Request* request) const;
 
+	bool getBoolean() const
+	{
+		fb_assert(litDesc.dsc_dtype == dtype_boolean);
+		return *reinterpret_cast<bool*>(litDesc.dsc_address);
```

Comment:
dtype_boolean is encoded in UCHAR, not `bool` (which has unspecified size).

```suggestion
		return *litDesc.dsc_address != '\0';
```